# Retail Demand & Market Intelligence Analytics

# Phase 2 – Data Understanding

## Business Objective

The objective of this project is to analyze historical retail transaction data to understand customer purchasing behavior, identify seasonal demand patterns, and ultimately compare those patterns with retail-sector stock performance.

The insights from this project will support business decisions related to inventory planning, customer segmentation, demand forecasting, and strategic planning.

## Import Required Libraries

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

pd.set_option("display.max_columns", None)
pd.set_option("display.max_rows", 20)

## Load the Dataset

In [3]:
df = pd.read_csv(
    "../data/raw/online_retail_II.csv",
    encoding="ISO-8859-1"
)
df.head()

,Invoice,StockCode,Description,Quantity,InvoiceDate,Price,Customer ID,Country
0,489434,85048,15CM CHRISTMAS GLASS BALL 20 LIGHTS,12,2009-12-01 07:45:00,6.95,13085.0,United Kingdom
1,489434,79323P,PINK CHERRY LIGHTS,12,2009-12-01 07:45:00,6.75,13085.0,United Kingdom
2,489434,79323W,WHITE CHERRY LIGHTS,12,2009-12-01 07:45:00,6.75,13085.0,United Kingdom
3,489434,22041,"RECORD FRAME 7"" SINGLE SIZE",48,2009-12-01 07:45:00,2.10,13085.0,United Kingdom
4,489434,21232,STRAWBERRY CERAMIC TRINKET BOX,24,2009-12-01 07:45:00,1.25,13085.0,United Kingdom


In [4]:
df.shape

(1067371, 8)

Observation: The dataset contains 1,067,371 transaction records across eight variables. This indicates that the dataset captures a large volume of retail activity over multiple years, making it suitable for analyzing customer purchasing behavior, seasonal demand patterns, and long-term business trends.

In [5]:
df.duplicated().sum()

np.int64(34335)

In [7]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1067371 entries, 0 to 1067370
Data columns (total 8 columns):
 #   Column       Non-Null Count    Dtype  
---  ------       --------------    -----  
 0   Invoice      1067371 non-null  str    
 1   StockCode    1067371 non-null  str    
 2   Description  1062989 non-null  str    
 3   Quantity     1067371 non-null  int64  
 4   InvoiceDate  1067371 non-null  str    
 5   Price        1067371 non-null  float64
 6   Customer ID  824364 non-null   float64
 7   Country      1067371 non-null  str    
dtypes: float64(2), int64(1), str(5)
memory usage: 65.1 MB
